operators/src/processing/raster_vector_join/non_aggregated.rs

```diff
@@ -93,61 +93,68 @@ where
+        let data = FeatureDataType::from(raster_processor.raster_data_type())
+            .null_feature_data(collection.len());
+        let collection = collection.add_column(new_column_name, data)?;
+        let collection_stream = once_stream(async move { Ok(collection) }).boxed();
+        Ok(collection_stream)
     }
 
     async fn process_typed_collection_chunk<'a, P: Pixel>(
         collection: FeatureCollection<G>,
         raster_processor: &'a dyn RasterQueryProcessor<RasterType = P>,
         new_column_name: &'a str,
-        query: VectorQueryRectangle,
+        query: RasterQueryRectangle,
         ctx: &'a dyn QueryContext,
         aggregation_method: FeatureAggregationMethod,
         ignore_no_data: bool,
     ) -> Result<BoxStream<'a, Result<FeatureCollection<G>>>> {
-        // make qrect smaller wrt. points
-        let bbox = collection
-            .bbox()
-            .and_then(|bbox| bbox.intersection(&query.spatial_bounds));
-
-        let time = collection
-            .time_bounds()
-            .and_then(|time| time.intersect(&query.time_interval));
-
-        if let (Some(q_bbox), Some(q_time)) = (bbox, time) {
-            let query = VectorQueryRectangle {
-                spatial_bounds: q_bbox,
-                time_interval: q_time,
-                spatial_resolution: query.spatial_resolution,
-            };
-
-            let raster_query = raster_processor.raster_query(query.into(), ctx).await?;
-
-            let collection = Arc::new(collection);
-
-            let collection_stream = raster_query
-                .time_multi_fold(
-                    move || Ok(VectorRasterJoiner::new(aggregation_method, ignore_no_data)),
-                    move |accum, raster| {
-                        let collection = collection.clone();
-                        async move {
-                            let accum = accum?;
-                            let raster = raster?;
-                            accum.extract_raster_values(&collection, &raster)
-                        }
-                    },
-                )
-                .map(move |accum| accum?.into_collection(new_column_name));
-
-            return Ok(collection_stream.boxed());
-        }
+        let raster_query = raster_processor.raster_query(query, ctx).await?;
+
+        let collection = Arc::new(collection);
+
+        let collection_stream = raster_query
+            .time_multi_fold(
+                move || Ok(VectorRasterJoiner::new(aggregation_method, ignore_no_data)),
+                move |accum, raster| {
+                    let collection = collection.clone();
+                    async move {
+                        let accum = accum?;
+                        let raster = raster?;
+                        accum.extract_raster_values(&collection, &raster)
+                    }
+                },
+            )
+            .map(move |accum| accum?.into_collection(new_column_name));
```

Comment:
Why did you (re)move the logic of modifying the boxes, etc.?

I would put it back since you did not win anything. In the block, where it is now, you handle two branches. Thus, it is better to read if you only see that I guess.

Reply:
it will at least save some generated code.  Building the query before we switch to the code path with generics will result in less generic code. Also in the macro we do a match block so there are also multiple branches.

While the empty collection will not be the default case, There will be many cases where there is no intersection in time and space.